### DISCUSSION TITLE
Could pundit authorization adapter pick up `Policy#permitted_attributes` automagically?

### DISCUSSION THREAD

**Discussion by gkop** [activeadmin/activeadmin#7867] [Ideas]
IE let's connect https://github.com/elabs/pundit#strong-parameters with our `ResourceDSL#permit_params`

**Comment by gkop** [comment]
I think this may be eminently doable, basically add some magic that achieves this behavior automatically:

``` ruby
ActiveAdmin.register Foo do
  permit_params do
    active_admin_authorization.retrieve_policy(resource_class).permitted_attributes
  end
end
```

**Comment by timoschilling** [comment]
We could provide something like this:

``` ruby
ActiveAdmin.register Foo do
  permit_params_by_pundit_policy
end
```

which is equal to yours

**Comment by gkop** [comment]
That would be nice!

**Comment by zharikovpro** [comment]
So far I use this hack in initializers/active_admin.rb

    # every resource must use permitted_attributes from Pundit policies
    ActiveSupport::Notifications.subscribe ActiveAdmin::Resource::RegisterEvent do |_event, resource|
      resource.dsl.send :permit_params do
        active_admin_authorization.retrieve_policy(resource_class).permitted_attributes
      end
    end

**Comment by cionescu** [comment]
For anyone looking to upgrade to 2.10 or above, this PR (https://github.com/activeadmin/activeadmin/pull/7262) changed the way Active Admin uses the Notifications API. The snippet above needs to be updated to:

```
  ActiveSupport::Notifications.subscribe ActiveAdmin::Resource::RegisterEvent do |name, start, finish, id, resource|
    resource.dsl.send :permit_params do
      active_admin_authorization.retrieve_policy(resource_class).permitted_attributes
    end
  end
```

**Comment by sjorsbranderhorst** [comment]
Hi cionescu, thanks for the snippet. When upgrading to 2.13 I had to rewrite it as:

```
    ActiveSupport::Notifications.subscribe ActiveAdmin::Resource::RegisterEvent do |name, start, finish, id, resource|
      resource[:active_admin_resource].dsl.send :permit_params do
        active_admin_authorization.retrieve_policy(resource_class).permitted_attributes
      end
    end
```
